addons/mail/models/mail_link_preview.py

```diff
@@ -0,0 +1,135 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from lxml import html, etree
+from urllib.parse import urlparse
+import requests
+
+
+class LinkPreview(models.Model):
+    _name = 'mail.link.preview'
+    _description = "Store link preview data"
+
+    message_id = fields.Many2one('mail.message', string='Message', index=True, ondelete='cascade')
+    guest_id = fields.Many2one('mail.guest', string="Guest")
+    og_type = fields.Char('type')
+    source_url = fields.Char('url', required=True)
+    og_title = fields.Char('title')
+    og_image = fields.Char('image')
+    og_description = fields.Text('description')
+    og_mimetype = fields.Char('mimetype')
+
+    # Some website are blocking non browser user agent. We make ourself like
+    # firefox to avoid issue.
+    request_headers = {
+        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; rv:91.0) Gecko/20100101 Firefox/91.0'
+    }
+
+    def _link_preview_format(self):
+        return [{
+            'description': preview.og_description,
+            'id': preview.id,
+            'image_url': preview.og_image,
+            'mimetype': preview.og_mimetype,
+            'title': preview.og_title,
+            'type': preview.og_type,
+            'url': preview.source_url,
+        } for preview in self]
+
+    @api.model
+    def _throttle_link_preview(self, url):
```

Comment:
method needs rename, it doesn't actually throttle anything, it just makes a check/returns a value